rfc/rfc005-distributed-network-using-grpc.md

```diff
@@ -0,0 +1,259 @@
+# RFC005 Distributed Network using gRPC
+
+|  |  |
+| :--- | :--- |
+| Nuts foundation | R.G. Krul |
+| Request for Comments: 005 | Nedap |
+|  | February 2021 |
+
+## Distributed Network using gRPC
+### Abstract
+
+### Status
+
+This document is currently a draft.
+
+### Copyright Notice
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1. Introduction
+
+When care organizations want to exchange public information (e.g. registry) using a decentralized network,
+they need a protocol that builds a distributed network allowing them to freely query and publish that information.
+This RFC describes a protocol for building such a distributed network using gRPC to achieve that goal.
+
+## 2. Terminology
+
+* **Node**: local Nuts software system acting (a.k.a. _Nuts Node_).
+* **Peer**: remote software system connected to the local node using the protocol described here.
+* **Transaction**: ...
+* **DAG** (Directed Acyclic Graph): ... 
+* **Heads**: Latest transactions of the DAG with no succeeding transactions that refer to it as previous transaction,
+
+## 3. Goals
+
+The protocol aims to synchronize the local DAG with that of peers by;
+
+1. making sure transactions produced by the local node are retrieved by its peers, and
+2. retrieving transactions produced by peers.
+
+## 4. Network Topology
+
+The network is a full mesh peer-to-peer network: all participants in the network try to connect to any peer they discover.
+
+The full mesh topology is expected to be performant up to ca. 20 nodes (source?). When a production network is expected
+surpass that number of nodes, the protocol should be adjusted to form a partial mesh where nodes only connect to a
+maximum number of peer. For instance, a IPFS node tries to connect to 10 other nodes which are randomly distributed.
+
+## 5. Operation
+
+The protocol generally operates as follows:
+
+1. The local node broadcasts their DAG state at a set interval:
+   - the references of the current day's (`T`) DAG head transactions.
+   - the block (see "Blocks") hashes of the previous 3 days (`T-1`, `T-2`, `T-3`). The hash of `T-3` is special because it also includes
```

Comment:
Changed the description a bit, since it's irrelevant whether it's calculated from the past blocks or transactions: both should lead to the same result since it's a XOR operation.